### ISSUE TITLE
Allow sorting of Tree like model list views

### ISSUE THREAD

**Issue by itdependsnetworks** [nautobot/nautobot#4811]
### Environment
* Nautobot version (Docker tag too if applicable): 2.0
* Python version: 3.11
* Database platform, version: psql
* Middleware(s): N/A

### Steps to Reproduce
1. Go to https://demo.nautobot.com/ipam/prefixes/
2. Click on headers
3.

### Expected Behavior

To sort based on the column header you created

### Observed Behavior

No sorting

**Comment by itdependsnetworks** [comment]
Seems like we just need to remove this https://github.com/nautobot/nautobot/blob/70012f7f8db4513e673c2061e07075a36b8f6580/nautobot/ipam/tables.py#L376, but not sure why it was there to begin with?

**Comment by glennmatthews** [comment]
Given that prefixes are hierarchical, I'm not sure sorting makes a lot of sense (similar to #1854). Can you give an example of the type of sorting you'd expect to be able to do here?

**Comment by lampwins** [comment]
IIRC in #1854 we chose at the time to disable sorting on tree models primarily due to the odd visual representation of nested data and because we did not have the cycles in 2.0 to properly solve that.

Let's use this issue to build a good story around the presentation and interaction with the Prefix hierarchy alone. I'm sure some of it will meaningfully carry over to other tree models like location, but there will be different use cases.

To start, I can see two primary views into the prefix data.

1. A collapsable, navigable tree. Filtering and sorting here, would only effect top level nodes in the tree.
2. A flat view that ignores the tree relationships in which all traditional filtering and sorting works.

As a user I would want a simple toggle between these two options in the list view.

I'm certain we can get more advanced from there.

In the first case, I care primary about drilling down into the data. In the second case, I care more about filtering and sorting, such as filtering by a location and then sorting by the role.

**Comment by lampwins** [comment]
For now, we will allow the sorting and when a user actually does sort, we will dynamically remove the hierarchy indentation.

**Comment by joewesch** [comment]
We had a request today regarding sorting in Locations, and I'm assuming it's missing due to the same reason. @lampwins do you want me to create a parallel issue?

**Comment by lampwins** [comment]
I don't think so. We would support sorting any column that it should be, in this model.

**Comment by gsnider2195** [comment]
Reopening this issue, needs to be implemented on the Prefix list view, and needs to be implemented when filtering a tree list view as well as sorting

**Comment by glennmatthews** [comment]
#5452 added sortability to the Prefix table. Remaining work now is to remove the indentation of entries whenever a filter is applied to any tree table.
